reference/constraints/When.rst

```diff
@@ -0,0 +1,202 @@
+When
+===========
+
+This constraint allows you to apply constraints validation only if the provided expression returns true.
```

Comment:
you should define a version added directive also
have a look at https://github.com/javiereguiluz/symfony-docs/commit/f32ee434c78f0210b1630aecf31b2c89a80f0074#diff-f552d75abf80f4985e72c4fe4d6664535dc95395a289456cf102f659bde0cfdfR4 as example